xarray/core/datatree.py

```diff
@@ -429,71 +512,85 @@ def parent(self: DataTree, new_parent: DataTree) -> None:
             raise ValueError("Cannot set an unnamed node as a child of another node")
         self._set_parent(new_parent, self.name)
 
+    def _to_dataset_view(self, rebuild_dims: bool) -> DatasetView:
+        variables = dict(self._data_variables)
+        variables |= self._coord_variables
+        if rebuild_dims:
+            dims = calculate_dimensions(variables)
+        else:
+            # Note: rebuild_dims=False can create technically invalid Dataset
+            # objects because it may not contain all dimensions on its direct
+            # member variables, e.g., consider:
+            #     tree = DataTree.from_dict(
+            #         {
+            #             "/": xr.Dataset({"a": (("x",), [1, 2])}),  # x has size 2
+            #             "/b/c": xr.Dataset({"d": (("x",), [3])}),  # x has size1
+            #         }
+            #     )
+            # However, they are fine for internal use cases, for align() or
+            # building a repr().
+            dims = dict(self._dims)
+        return DatasetView._constructor(
+            variables=variables,
+            coord_names=set(self._coord_variables),
+            dims=dims,
+            attrs=self._attrs,
+            indexes=dict(self._indexes),
+            encoding=self._encoding,
+            close=None,
+        )
+
     @property
     def ds(self) -> DatasetView:
         """
         An immutable Dataset-like view onto the data in this node.
 
-        For a mutable Dataset containing the same data as in this node, use `.to_dataset()` instead.
+        For a mutable Dataset containing the same data as in this node, use
+        `.to_dataset()` instead.
```

Comment:
can we mention here that this will include the inherited variables?

Reply:
good point, done!